governance/program/src/state/proposal.rs

```diff
@@ -229,25 +280,84 @@ impl Proposal {
     fn get_final_vote_state(
         &mut self,
         max_vote_weight: u64,
         config: &GovernanceConfig,
-    ) -> ProposalState {
-        let yes_vote_threshold_count =
-            get_yes_vote_threshold_count(&config.vote_threshold_percentage, max_vote_weight)
+    ) -> Result<ProposalState, ProgramError> {
+        // Get the min vote weight required for options to pass
+        let min_vote_threshold_weight =
+            get_min_vote_threshold_weight(&config.vote_threshold_percentage, max_vote_weight)
                 .unwrap();
 
-        // Yes vote must be equal or above the required yes_vote_threshold_percentage and higher than No vote
-        // The same number of Yes and No votes is a tie and resolved as Defeated
-        // In other words  +1 vote as a tie breaker is required to Succeed
-        if self.yes_votes_count >= yes_vote_threshold_count
-            && self.yes_votes_count > self.no_votes_count
-        {
-            ProposalState::Succeeded
-        } else {
+        // If the proposal has a reject option then any other option must beat it regardless of the configured min_vote_threshold_weight
+        let deny_vote_weight = self.deny_vote_weight.unwrap_or(0);
+
+        let mut best_succeeded_option_weight = 0;
+        let mut best_succeeded_option_count = 0;
+
+        for option in self.options.iter_mut() {
+            // Any positive vote (Yes) must be equal or above the required min_vote_threshold_weight and higher than the reject option vote (No)
+            // The same number of positive (Yes) and rejecting (No) votes is a tie and resolved as Defeated
+            // In other words  +1 vote as a tie breaker is required to succeed for the positive option vote
+            if option.vote_weight >= min_vote_threshold_weight
+                && option.vote_weight > deny_vote_weight
+            {
+                option.vote_result = OptionVoteResult::Succeeded;
+
+                match option.vote_weight.cmp(&best_succeeded_option_weight) {
+                    Ordering::Greater => {
+                        best_succeeded_option_weight = option.vote_weight;
+                        best_succeeded_option_count = 1;
+                    }
+                    Ordering::Equal => best_succeeded_option_count += 1,
+                    Ordering::Less => {}
+                }
+            } else {
+                option.vote_result = OptionVoteResult::Defeated;
+            }
+        }
+
+        let mut final_state = if best_succeeded_option_count == 0 {
+            // If none of the individual options succeeded then the proposal as a whole is defeated
             ProposalState::Defeated
+        } else {
+            match self.vote_type {
+                VoteType::SingleChoice => {
+                    let proposal_state = if best_succeeded_option_count > 1 {
+                        // If there is more than one winning option then the single choice proposal is considered as defeated
```

Comment:
For a SingleChoice proposal there can only be a single winning option and this is why `best_succeeded_option_count` is calculated. For example the following vote would have `best_succeeded_option_count=2` and be defeated because both A and B can't be winners at the same time
A: 40%
B: 40%
Nay: 20%

Reply:
Ah I see, I misread the code, makes sense!